### PR TITLE
start adding fancier model

### PR DESCRIPTION
**Pull request by avivajpeyi**
**Convolutional Layers:** 

- Since data is 1D (length 256--> 1D convolutional layers in place of or in addition to dense layers. 
- _Why:_ Convolutions can better capture local dependencies and may result in a more compact representation.

**Deeper Networks or Residual Connections:**

- Adding extra layers in both the encoder and decoder can help learn more expressive features. 
- BUT I need to be careful about overfitting... 

**Normalization and Regularization:**

- Batch normalization or layer normalization can stabilize training. 
- Dropout or L2 weight regularization might also help, especially with only ~1000 training examples.

**Activation Functions:**

- I used ReLU, but LeakyReLU or ELU might help --> supposedly help preventing dead neurons.

### CODE REVIEW

**Comment by avivajpeyi** [comment]
Here are plots from training with Z=32 (new model) [took about 25 min]

loss
reconstruction_E1000
distributions_E1000

**Comment by avivajpeyi** [comment]
With the old model (no droupouts/conv/leakyrelu) [2min]

reconstruction_E1000
distributions_E1000
loss
